src/movies.js

```diff
@@ -1,25 +1,131 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const directors = moviesArray.map(el => el.director);
+    return directors;                      
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    const spielbergMovies = moviesArray.filter(el => el.director === "Steven Spielberg" && el.genre.includes("Drama"));
+    return spielbergMovies.length;
+ }
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (moviesArray.length === 0) {
+        return 0;
+    }
+    const notableMovies = moviesArray.reduce((acc,el) => {
+        if (typeof el.score === 'number') {
+        return acc + el.score;
+        } else {
+        return acc;
+        }
+     },0);
+     const scoreNumber = notableMovies / moviesArray.length;
+     const roundedAverage = Math.round(scoreNumber* 100) / 100;
+     return roundedAverage;
+    } 
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+    const dramaMovies = moviesArray.filter(el => el.genre.includes("Drama"));
+    console.log(dramaMovies);
+    if (dramaMovies.length === 0) { 
+        return 0;
+    } 
+    const totalDramaMovies = dramaMovies.reduce((acc, el) => {
+        if (typeof el.score === 'number') {
+            return acc + el.score;
+            } else {
+            return acc;
+            }
+    },0);
+    const dramaMovieAcc = totalDramaMovies / dramaMovies.length;
+    const roundedAverage = Math.round(dramaMovieAcc* 100) / 100;
+    return roundedAverage;
```

Comment:
Esto me suena que lo he visto antes! :D Esta logica la tienes en el scoreAverage, entonces para no repetir codigo y usar bien la abstracción vamos a cogerlo!!

Te pongo refactor
```suggestion
    return scoreAverage(dramaMovies);
```